dagfactory/dagbuilder.py

```diff
@@ -726,6 +726,85 @@ def process_file_with_datasets(file: str, datasets_conditions: str) -> Any:
             datasets_uri = utils.get_datasets_uri_yaml_file(file, list(dataset_map.keys()))
             return [Dataset(uri) for uri in datasets_uri]
 
+    @staticmethod
+    def _asset_schedule(value):
+        from functools import reduce
+
+        from airflow.sdk import Asset
+
+        def _init_asset(asset_dict: dict) -> Asset:
+            if "watchers" in asset_dict:
+                watchers_list = []
+                for watcher in asset_dict.get("watchers"):
+                    watcher_class = import_string(watcher.get("callable"))
+                    trigger_class = import_string(watcher.get("trigger").get("callable"))
+                    trigger_class_params = watcher.get("trigger").get("params")
+                    watchers_list.append(
+                        watcher_class(name=watcher.get("name"), trigger=trigger_class(**trigger_class_params))
+                    )
+                asset_dict["watchers"] = watchers_list
+            return Asset(**asset_dict)
+
+        def _combine_assets(assets, op):
+            if op == "or":
+                return reduce(lambda a, b: a | b, assets)
+            elif op == "and":
+                return reduce(lambda a, b: a & b, assets)
+            else:
+                raise ValueError(f"Unknown operator: {op}")
+
+        if isinstance(value, dict):
+            if "or" in value:
+                return _combine_assets([DagBuilder._asset_schedule(item) for item in value["or"]], op="or")
```

Comment:
Python has [known performance issues](https://www.designgurus.io/answers/detail/why-is-python-recursion-so-expensive-and-what-can-we-do-about-it) with recursion. Would there be an alternative approach to implement this?